src/transformers/models/seggpt/modeling_seggpt.py

```diff
@@ -0,0 +1,732 @@
+
+        self.decoder = SegGPTDecoder(config)
+        self.num_patches = (config.image_size[0] // config.patch_size) * (config.image_size[1] // config.patch_size)
+
+        self.apply(self._init_weights)
+        self.patch_size = config.patch_size
+
+    def patchify(self, imgs):
+        """
+        imgs: (N, 3, H, W) x: (N, L, patch_size**2 *3)
+        """
+        p = self.patch_size
+        assert imgs.shape[2] == 2 * imgs.shape[3] and imgs.shape[2] % p == 0
+
+        w = imgs.shape[3] // p
+        h = w * 2
+        x = imgs.reshape(shape=(imgs.shape[0], 3, h, p, w, p))
+        x = torch.einsum("nchpwq->nhwpqc", x)
+        x = x.reshape(shape=(imgs.shape[0], h * w, p**2 * 3))
+        return x
+
+    def unpatchify(self, x):
+        """
+        x: (N, L, patch_size**2 *3) imgs: (N, 3, H, W)
+        """
+        p = self.patch_size
+        w = int((x.shape[1] * 0.5) ** 0.5)
+        h = w * 2
+        assert h * w == x.shape[1]
+
+        x = x.reshape(shape=(x.shape[0], h, w, p, p, 3))
+        x = torch.einsum("nhwpqc->nchpwq", x)
+        imgs = x.reshape(shape=(x.shape[0], 3, h * p, w * p))
+        return imgs
+
+    def forward_decoder(self, x):
+        x = self.decoder_embed(x)  # BxhxwxC
+        p = self.patch_size
+        h, w = x.shape[1], x.shape[2]
+        x = x.reshape(shape=(x.shape[0], h, w, p, p, self.decoder_embed_dim))
+        x = torch.einsum("nhwpqc->nchpwq", x)
+        x = x.reshape(shape=(x.shape[0], -1, h * p, w * p))
+
+        x = self.decoder_pred(x)  # Bx3xHxW
+        return x
+
+    def forward(
+        self,
+        pixel_values,
+        prompt_pixel_values,
+        seg_type=None,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+    ):
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+        output_attentions = output_attentions if output_attentions is not None else self.config.output_attentions
+        output_hidden_states = (
+            output_hidden_states if output_hidden_states is not None else self.config.output_hidden_states
+        )
+
+        encoder_outputs = self.encoder(
+            pixel_values,
+            prompt_pixel_values,
+            seg_type,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+        )
+
+        decoder_output = self.decoder(encoder_outputs[0])
+        logits = self.patchify(decoder_output)  # [N, L, p*p*3]
+
+        if not return_dict:
+            outputs = (
+                logits,
+                encoder_outputs[0],
+            )
+            outputs = outputs + (encoder_outputs[1],) if output_hidden_states else outputs
+            return outputs + (encoder_outputs[-1],) if output_attentions else outputs
+
+        return SegGPTModelOutput(
+            logits=logits,
```

Comment:
base models should not output logits. Only head models.

Reply:
Understood, Done.